packages/dht/test/integration/ConnectionManager.test.ts

```diff
@@ -0,0 +1,183 @@
+import { ConnectionManager } from "../../src/connection/ConnectionManager"
+import { Event as ITransportEvent } from "../../src/transport/ITransport"
+import { Message, MessageType, NodeType, PeerDescriptor } from "../../src/proto/DhtRpc"
+import { createPeerDescriptor } from '../utils'
+import { waitForEvent } from 'streamr-test-utils'
+import { Event as ConnectionEvent } from '../../src/connection/IConnection'
+import { ClientWebSocket } from '../../src/connection/WebSocket/ClientWebSocket'
+import { MockConnectionManager } from '../../src/connection/MockConnectionManager'
+import { PeerID } from '../../src/helpers/PeerID'
+import { Simulator } from '../../src/connection/Simulator'
+
+describe('ConnectionManager', () => {
+    const appId = 'demo'
+
+    const mockPeerDescriptor1: PeerDescriptor = {
+        peerId: PeerID.fromString("tester1").value,
+        type: NodeType.NODEJS
+    }
+    const mockPeerDescriptor2: PeerDescriptor = {
+        peerId: PeerID.fromString("tester2").value,
+        type: NodeType.NODEJS
+    }
+    const simulator = new Simulator()
+
+    const mockTransport = new MockConnectionManager(mockPeerDescriptor1, simulator)
+    const mockConnectorTransport1 = new MockConnectionManager(mockPeerDescriptor1, simulator)
+    const mockConnectorTransport2 = new MockConnectionManager(mockPeerDescriptor2, simulator)
+
+    beforeAll(async () => {
+
+    })
+
+    it('Can start alone', async () => {
+        const connectionManager = new ConnectionManager({ transportLayer: mockTransport, webSocketHost: 'localhost', webSocketPort: 9991 })
+
+        const result = await connectionManager.start()
+        expect(result.ip).toEqual('localhost')
+        expect(result.openInternet).toEqual(true)
+
+        await connectionManager.stop()
+    })
+
+    it('Throws an async exception if fails to connect to entrypoints', async () => {
+
+        const connectionManager = new ConnectionManager({
+            transportLayer: mockTransport,
+            webSocketPort: 9992, entryPoints: [
+                { peerId: Uint8Array.from([1, 2, 3]), type: NodeType.NODEJS, websocket: { ip: 'localhost', port: 123 } }
+            ]
+        })
+        
+        await expect(connectionManager.start())
+            .rejects
+            .toThrow('Failed to connect to the entrypoints')
+
+        await connectionManager.stop()
+    })
+
+    it('Can probe connectivity in open internet', async () => {
+        const connectionManager = new ConnectionManager({ transportLayer: mockTransport, webSocketHost: 'localhost', webSocketPort: 9993 })
```

Comment:
```suggestion
        const connectionManager1 = new ConnectionManager({ transportLayer: mockTransport, webSocketHost: 'localhost', webSocketPort: 9993 })
```

Whenever there are multiple instances, it would be informative if we have all instances numbered. It tells the intention that there will be more instances. Of course if some good non-numeric names would be applicable, that would be even better.